thrust/system/cuda/detail/future.inl

```diff
@@ -152,7 +152,7 @@ struct stream_deleter final
 struct stream_conditional_deleter final
 {
 private:
-  bool const cond_;
```

Comment:
Why is this part of the change needed?

Reply:
Const member variables disable default move semantics:

https://lesleylai.info/en/const-and-reference-member-variables/

It seems like trouble to stick that in a unique_ptr.

Reply:
Ah, so it's because it's used as a deleter (I should probably pay more attention to names :D).